### ISSUE TITLE
Allow adding substituters in devenv.yaml

### ISSUE THREAD

**Issue by bobvanderlinden** [cachix/devenv#444]
Currently it is not possible to add additional substituters to the `devenv.yaml` or `flake.nix`. It forces the repository owner to let developers know that they need to run `cachix use {cachix repo}` through a readme.

Nix flakes already has support for setting `nixConfig`, which is already used by devenv itself. It would be nice if we could add additional entries using configuration in `devenv.yaml`.

There are a number of options to model this:

Combine with `inputs`, so that READMEs can just tell to add a single block of input to their `devenv.yaml`:

```yaml
inputs:
  nixpkgs-ruby:
    url: https://github.com/bobvanderlinden/nixpkgs-ruby
    substituters:
      - https://nixpkgs-ruby.cachix.org
    public_keys:
      - "nixpkgs-ruby.cachix.org-1:vrcdi50fTolOxWCZZkw0jakOnUI1T19oYJ+PRYdK4SM="
```

Add a separate section with substituters:

```yaml
substituters:
- url: https://nixpkgs-ruby.cachix.org
  public_key: "nixpkgs-ruby.cachix.org-1:vrcdi50fTolOxWCZZkw0jakOnUI1T19oYJ+PRYdK4SM="
```

Scrap the whole concept of substituters and call is binary caches:

```yaml
caches:
- url: https://nixpkgs-ruby.cachix.org
  public_key: "nixpkgs-ruby.cachix.org-1:vrcdi50fTolOxWCZZkw0jakOnUI1T19oYJ+PRYdK4SM="
```

What also could be considered, but I imagine is a bit trickier, is to inherit `nixConfig` and/or cache configuration from `devenv.yaml` inputs. That way whenever `nixpkgs-ruby` changes its substituters and people update `nixpkgs-ruby` input, they will automatically be asked to add the new binary cache.

Currently the `nixConfg` that is in [`flake.nix`](https://github.com/bobvanderlinden/nixpkgs-ruby/blob/3319e158e168ddcac5e932e00884942a67a5979e/flake.nix#L7-L10) is defunct when not directly using `flake.nix`.

Note that I'm using nixpkgs-ruby as an example here because it is a convenient public repository and binary cache, but I actually ran into this issue because I wanted coworkers to use a private Cachix repository.

**Comment by domenkozar** [comment]
Similar to #436

**Comment by bobvanderlinden** [comment]
Right, I'll consider this merged into #436
